File: streamdeck-plugin/src/event_handlers/stop_presenting_screen_handler.py
from event_handlers.base_event_handler import EventHandler


class StopPresentingScreenEventHandler(EventHandler):
    """
    A Stream Deck button that stop presenting the screen.
    """

    STREAM_DECK_ACTION = "com.chrisregado.googlemeet.stoppresentingscreen"

    async def _key_up_handler(self, event: dict) -> None:
        leave = self._make_simple_sd_event("stopPresentingScreen")
        await self._browser_manager.send_to_clients(leave)

File: browser-extension/event_handlers/stop_presenting_screen_handler.js
class StopPresentingScreenEventHandler extends SDEventHandler {

  handleStreamDeckEvent = (message) => {
    if (message.event === "stopPresentingScreen") {
      this._stopPresentingScreen();
    }
  }

  _stopPresentingScreen = () => {
    const button = document.querySelector('[jsname="aK5XXd"]');
    if (!button) {
      throw new ControlsNotFoundError("No stop presenting screen button found!")
    }
    button.click();
  }

}
